utils/frame/benchmarking-cli/src/block/bench.rs

```diff
@@ -0,0 +1,240 @@
+// This file is part of Substrate.
+
+// Copyright (C) 2022 Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use sc_block_builder::BlockBuilderApi;
+use sc_cli::Result;
+use sc_client_api::{BlockBackend, UsageProvider};
+use sp_api::{ApiExt, BlockId, HeaderT, ProvideRuntimeApi};
+use sp_blockchain::HeaderBackend;
+use sp_runtime::{traits::Block as BlockT, DigestItem};
+
+use clap::Args;
+use log::{info, warn};
+use serde::Serialize;
+use std::{marker::PhantomData, sync::Arc, time::Instant};
+
+use crate::storage::record::Stats;
+
+/// Parameters to configure a block benchmark.
+#[derive(Debug, Default, Serialize, Clone, PartialEq, Args)]
+pub struct BenchmarkParams {
+	/// Skip benchmarking NO-OP extrinsics.
+	#[clap(long)]
+	pub skip_extrinsic: bool,
+
+	/// Skip benchmark an empty block.
+	#[clap(long)]
+	pub skip_block: bool,
+
+	/// Specify the id of a full block. 0 is treated as last block.
+	/// The block should be filled with `System::remark` extrinsics.
+	#[clap(long, default_value = "0")]
+	pub full_block: u32,
+
+	/// Specify the id of an empty block. 0 is treated as last block.
+	/// The block should not contains any user extrinsics but only inherents.
+	#[clap(long, default_value = "0")]
+	pub empty_block: u32,
+
+	/// How many executions should be measured.
+	#[clap(long, default_value = "100")]
+	pub repeat: u32,
+
+	/// How many executions should be run as warmup.
+	#[clap(long, default_value = "100")]
+	pub warmup: u32,
+
+	/// Maximum number of inherents that can be present in a block
+	/// such that the block will still be considered empty.
+	///
+	/// Default is 1 since that is the case for Substrate.
+	/// This check exists to make sure that a non-empty block is not
+	/// accidentally counted as empty.
+	#[clap(long, default_value = "1")]
+	pub max_inherents: u32,
+
+	/// Minimum number of extrinsics that must be present in a block
+	/// such that the block will be considered full.
+	///
+	/// Default is 12_000 since in Substrate a block can hold that many NO-OPs.
+	#[clap(long, default_value = "12000")]
+	pub min_extrinsics: u32,
```

Comment:
this seems strange. blocks can be full for other extrinsics which are less than this. I dont think hardcoding a number here makes sense. If you want to check the block is full, you should look at the weight of the block.

Reply:
Yea makes more sense. I will replace it with some kind of `--min-weight <PERCENT>`.